larq_compute_engine/mlir/python/converter.py

```diff
@@ -37,46 +39,18 @@ def concrete_function_from_keras_model(model):
     return func.get_concrete_function()
 
 
-def convert_keras_model(model: tf.keras.Model) -> bytes:
-    """Converts a Keras model to TFLite flatbuffer.
-
-    !!! example
-        ```python
-        tflite_model = convert_keras_model(model)
-        with open("/tmp/my_model.tflite", "wb") as f:
-            f.write(tflite_model)
-        ```
-
-    # Arguments
-    model: A [`tf.keras.Model`](https://www.tensorflow.org/api_docs/python/tf/keras/Model) to convert.
-
-    # Returns
-    The converted data in serialized format.
-    """
-    if not tf.executing_eagerly():
-        raise RuntimeError(
-            "Graph mode is not supported. Please enable eager execution using "
-            "tf.enable_eager_execution() when using TensorFlow 1.x"
-        )
-    func = concrete_function_from_keras_model(model)
-    if version.parse(tf.__version__) >= version.parse("1.15"):
-        frozen_func = convert_variables_to_constants_v2(func, lower_control_flow=False)
-    else:
-        frozen_func = convert_variables_to_constants_v2(func)
-    input_tensors = [
-        tensor for tensor in frozen_func.inputs if tensor.dtype != tf.dtypes.resource
-    ]
-    output_tensors = frozen_func.outputs
-
-    graph_def = frozen_func.graph.as_graph_def()
+def convert_tensorflow_graph(
+    graph: FuncGraph, input_tensors: List[tf.Tensor], output_tensors: List[tf.Tensor]
+):
+    graph_def = graph.as_graph_def()
     # Run a constant folding using grappler since we currently don't implement
     # folding for LCE custom ops
     graph_def = run_graph_optimizations(
         graph_def,
         input_tensors,
         output_tensors,
         config=get_grappler_config(["constfold"]),
-        graph=frozen_func.graph,
+        graph=graph,
     )
```

Comment:
Do we need these optimizations? Removing them would allow even more flexibility, as we could pass in the graph def instead.